upload/system/library/Packetery/Vendor/AddEditPage.php

```diff
@@ -97,45 +95,26 @@ public function getAddVendorFormValidationErrors(array $formData) {
 	 */
 	private function validateWeightRules(array $weightRules) {
 		$errors = [];
-
-		foreach ($weightRules as $rule) {
-			if (!isset($rule['max_weight']) || !is_numeric($rule['max_weight']) || $rule['max_weight'] <= 0) {
-				$errors[] = 'vendor_add_error_rule_max_weight_invalid';
+		$weights = [];
+
+		foreach ($weightRules as $counter => $rule) {
+			if (!is_numeric($rule['max_weight']) || $rule['max_weight'] <= 0) {
+				$errors[$counter]['max_weight'] = $this->language->get('vendor_add_error_rule_max_weight_invalid');
+			} else {
+				if (in_array($rule['max_weight'], $weights, true)) {
+					$errors[$counter]['max_weight'] = $this->language->get('vendor_add_error_rule_duplicate_weights');
+				}
+				$weights[] = $rule['max_weight'];
 			}
 
-			if (!isset($rule['price']) || !is_numeric($rule['price']) || $rule['price'] <= 0) {
-				$errors[] = 'vendor_add_error_rule_price_invalid';
+			if (!is_numeric($rule['price']) || $rule['price'] <= 0) {
```

Comment:
nechceme povolit cenu 0?
váhu 0 ne, to chápu, ale třeba chce dopravu zadarmo pro váhu do nějakého limitu... Zeptej se Jardy, prosím.